django/core/handlers/middleware.py

```diff
@@ -0,0 +1,10 @@
+class MiddlewareMixin(object):
+    def __call__(self, request, response_factory):
+        response = None
+        if hasattr(self, 'process_request'):
+            response = self.process_request(request)
+        if not response:
+            response = response_factory(request)
```

Comment:
For completeness, shouldn't there also be a try/except here and a call to `process_exception` if an exception occurs?